synapseclient/table.py

```diff
@@ -829,75 +857,155 @@ class Dataset(SchemaBase):
     :py:classmethod:`synapseclient.client.create_snapshot_version`.
 
     Example::
+
         syn = synapseclient.login()
         syn.create_snapshot_version(
             dataset.id,
             label="v1.0",
             comment="This is version 1")
     """
     _synapse_entity_type: str = "org.sagebionetworks.repo.model.table.Dataset"
-    _property_keys: List[str] = SchemaBase._property_keys + ['dataset_items']
+    _property_keys: List[str] = SchemaBase._property_keys + ['datasetItems']
+    _local_keys: List[str] = SchemaBase._local_keys + ['folders_to_add', 'force']
 
     def __init__(self, name=None, columns=None, parent=None, properties=None,
-                 annotations=None, local_state=None, dataset_items=None, **kwargs):
-        self.properties.setdefault('dataset_items', [])
+                 annotations=None, local_state=None, dataset_items=None,
+                 folder_ids=None, force=None, **kwargs):
+        self.properties.setdefault('datasetItems', [])
+        self.__dict__.setdefault('folders_to_add', set())
```

Comment:
Why the difference between `folders_to_add` and `folder_ids`

Reply:
Mostly because `folder_ids` is a shorter variable for users to write :upside_down_face:

Reply:
`folders` is even shorter, so maybe we should use that instead, heh.

Reply:
I like `folders`